src/Amadeus.Server/Services/UserService.cs

```
@@ -0,0 +1,113 @@
+		}
+
+
+		/// <summary>
+		/// Get all the users in db.
+		/// </summary>
+		/// <returns>All the users in the db.</returns>
+		public Task<List<User>> GetAll()
+		{
+			return _context.Users
+				.AsNoTracking()
+				.ToListAsync();
+		}
+
+		/// <summary>
+		/// Get a user by it's id.
+		/// </summary>
+		/// <param name="id">The id of the user to get.</param>
+		/// <returns>The user corresponding at the id.</returns>
+		public Task<User> GetUserById(int id)
+		{
+			return _context.Users.AsNoTracking().FirstOrDefaultAsync(x => x.Id == id);
+		}
+
+		/// <summary>
+		/// Create saves the user in the db.
+		/// </summary>
+		/// <param name="user">The user to save in the db.</param>
+		/// <returns>The user saved in db.</returns>
+		public async Task<User> Create(User user)
+		{
+			Debug.Assert(user != null, nameof(user) + " != null");
+			user.CreatedAt = DateTime.UtcNow;
+
+			// generate a 128-bit salt using a cryptographically strong random sequence of nonzero values
+			byte[] salt = new byte[128 / 8];
+			using (var rngCsp = new RNGCryptoServiceProvider())
+			{
+				rngCsp.GetNonZeroBytes(salt);
+			}
+
+			// derive a 256-bit subkey (use HMACSHA256 with 100,000 iterations)
+			user.Password = Convert.ToBase64String(KeyDerivation.Pbkdf2(
+				password: user.Password,
+				salt: salt,
+				prf: KeyDerivationPrf.HMACSHA256,
+				iterationCount: 100000,
+				numBytesRequested: 256 / 8));
+
+			_context.Add(user);
+			try
+			{
+				await _context.SaveChangesAsync();
+			}
+			catch (Microsoft.EntityFrameworkCore.DbUpdateException exception)
+			{
+				throw new ArgumentException("User with fields like this already exists");
+			}
+
+			return user;
+		}
+
+		/// <summary>
+		/// Delete a user by id.
+		/// </summary>
+		/// <param name="id">id of the user to delete.</param>
+		/// <returns>True if the user was deleted false otherwise</returns>
+		public async Task<User> Delete(int id)
+		{
+			User user = await GetUserById(id);
+
+			if (user != null)
```

You should throw an exception if the user could not be deleted/found. Here you will return a 200 instead of a 404.
The method Delete should return a Task and not a Task<User> since the given user has been deleted. We don't want the interface to force fetch the user before deleting it.